server/controllers/posts.js

```diff
@@ -25,7 +26,35 @@ const savePost = asyncMiddleware(async (req, res, next) => {
   res.status(200).json({ success: true })
 })
 
+
+const searchPosts = function(req, res) {
+  if (!req.query.tag) {
+    res.status(400).send('no tag specified!')
+    return
+  }
+  const value  = req.query.tag
+
+  let postsDocsPromise = searchService.findByTag(value)
+  postsDocsPromise.then(snapshot => {
+    if (snapshot.empty) {
+      console.log('No matching documents.');
+      return;
```

Comment:
You need to return a response here as well in case of empty results. And in rejected promise too.

Reply:
Changed to `return res.status(400).send('No matching documents.')`
What do you mean in rejected promise too, because I have .catch in promise? Can you show an example?